bitcoin/src/crypto/key.rs

```diff
@@ -233,22 +233,22 @@ impl fmt::Display for PublicKey {
 impl FromStr for PublicKey {
     type Err = ParsePublicKeyError;
     fn from_str(s: &str) -> Result<PublicKey, ParsePublicKeyError> {
+        use HexToArrayError::*;
+
         match s.len() {
             66 => {
-                PublicKey::from_slice(&<[u8; 33]>::from_hex(s).map_err(|op| {
-                    match op {
-                        HexToArrayError::Conversion(HexToBytesError::InvalidChar(char)) => ParsePublicKeyError::InvalidChar(char),
-                        HexToArrayError::Conversion(HexToBytesError::OddLengthString(_)) | HexToArrayError::InvalidLength(_,_) => unreachable!("invalid length"),
-                    }
-                })?).map_err(From::from)
+                let bytes = <[u8; 33]>::from_hex(s).map_err(|e| match e {
+                    InvalidChar(e) => ParsePublicKeyError::InvalidChar(e.invalid_char()),
+                    InvalidLength(_) => unreachable!("length checked already")
+                })?;
+                Ok(PublicKey::from_slice(&bytes).expect("length checked already"))
```

Comment:
This will panic on invalid keys. Not all 66 bytes hex strings are valid keys.

Reply:
Nice reviewing man! Caught a bug I was introducing, legend. Fixed now to use `?` (instead of the original `map_err(From::from)`)

Reply:
We should have failing test case PublicKey::from_str for invalid keys. Making an issue for this

Reply:
Opened a [pr](https://github.com/rust-bitcoin/rust-bitcoin/pull/2563) for the same.